tests/unit/src/test/scala/tests/NewFilesLspSuite.scala

```diff
@@ -19,26 +19,84 @@ class NewFilesLspSuite extends BaseLspSuite("new-files") {
   override def initializationOptions: Option[InitializationOptions] =
     Some(InitializationOptions.Default.copy(inputBoxProvider = Some(true)))
 
-  check("new-worksheet")(
+  check("new-worksheet-picked")(
     Some("a/src/main/scala/"),
-    "worksheet",
-    Some("Foo"),
+    Right(worksheet),
+    Right("Foo"),
     "a/src/main/scala/Foo.worksheet.sc",
     ""
   )
 
-  check("new-ammonite-script")(
+  check("new-worksheet-name-provided")(
     Some("a/src/main/scala/"),
-    "ammonite",
-    Some("Foo"),
+    Left(worksheet),
+    Right("Foo"),
+    "a/src/main/scala/Foo.worksheet.sc",
+    ""
+  )
+
+  check("new-worksheet-fully-provided")(
+    Some("a/src/main/scala/"),
+    Left(worksheet),
```

Comment:
Could you also use named parameters here and in the method above?

Reply:
For sure. Added.